openapi-validator/src/main/java/io/ballerina/openapi/validator/PreValidator.java

```diff
@@ -0,0 +1,279 @@
+import static io.ballerina.openapi.validator.Constants.ATTRIBUTE_EXCLUDE_OPERATIONS;
+import static io.ballerina.openapi.validator.Constants.ATTRIBUTE_EXCLUDE_TAGS;
+import static io.ballerina.openapi.validator.Constants.ATTRIBUTE_FAIL_ON_ERRORS;
+import static io.ballerina.openapi.validator.Constants.ATTRIBUTE_OPERATIONS;
+import static io.ballerina.openapi.validator.Constants.EMBED;
+import static io.ballerina.openapi.validator.Constants.OPENAPI_ANNOTATION;
+import static io.ballerina.openapi.validator.Constants.TRUE;
+import static io.ballerina.openapi.validator.ValidatorUtils.isHttpService;
+import static io.ballerina.openapi.validator.ValidatorUtils.parseOpenAPIFile;
+import static io.ballerina.openapi.validator.ValidatorUtils.reportDiagnostic;
+import static io.ballerina.openapi.validator.error.CompilationError.EMPTY_CONTRACT_PATH;
+import static io.ballerina.openapi.validator.error.CompilationError.NON_HTTP_SERVICE;
+import static io.ballerina.openapi.validator.error.CompilationError.UNEXPECTED_EXCEPTIONS;
+
+/**
+ * This PreValidator class contains validation for given service if it is a http service and summaries all the
+ * resources.
+ *
+ * @since 1.1.0
+ */
+public class PreValidator implements Validator {
+    private Filter filter;
+    private SyntaxNodeAnalysisContext context;
+    private OpenAPI openAPI;
+    public void initialize(SyntaxNodeAnalysisContext context) {
+        this.context = context;
+        this.openAPI = null;
+    }
+    public Filter getFilter() {
+        return filter;
+    }
+
+    public OpenAPI getOpenAPI() {
+        return openAPI;
+    }
+
+    /**
+     * During the Pre validation , it is checking whether the given service is http service, that service has openapi
+     * annotation @openapi:ServiceInfo and that annotation includes valid openapi contract path.
+     *
+     */
+    @Override
+    public void validate() {
+        // 1. Checking receive service node has compilation issue
+        SemanticModel semanticModel = context.semanticModel();
+        List<Diagnostic> diagnostics = semanticModel.diagnostics();
+        boolean erroneousCompilation = diagnostics.stream()
+                .anyMatch(d -> DiagnosticSeverity.ERROR == d.diagnosticInfo().severity());
+        if (erroneousCompilation) {
+            return;
+        }
+
+        ServiceDeclarationNode serviceNode = (ServiceDeclarationNode) this.context.node();
+        Path ballerinaFilePath = getBallerinaFilePath();
+        Optional<MetadataNode> metadata = serviceNode.metadata();
+        boolean validatorEnable = false;
+        Location location = serviceNode.location();
+
+        // 2.Test given service is http service, if not this will return WARNING and execute the ballerina
+        // file.
+        if (!isHttpService(serviceNode, context.semanticModel())) {
+            reportDiagnostic(context, NON_HTTP_SERVICE, location, DiagnosticSeverity.WARNING);
+            return;
+        }
```

Comment:
Better to refactor these calculations in order to avoid unnecessary processing. (i.e. if we do the http service check just after getting the service node, we dont have to compute `ballerinaFilePath` and `metadata`)

Reply:
I will change the process, the necessary calculations before it use instead of early assigned.